asserts/database.go

```diff
@@ -176,3 +223,101 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+// Add persists the assertions after ensuring it is properly signed and consistent with all the stored knowledge.
+// Returns ErrAssertionNotSuperseding if trying to add a previous revision of the assertion.
+func (db *Database) Add(assert Assertion) error {
+	reg, err := checkAssertType(assert.Type())
+	if err != nil {
+		return err
+	}
+	err = db.Check(assert)
+	if err != nil {
+		return err
+	}
+	primaryKey := make([]string, len(reg.primaryKey))
+	for i, k := range reg.primaryKey {
+		keyVal := assert.Header(k)
+		if keyVal == "" {
+			return fmt.Errorf("missing primary key header: %v", k)
+		}
+		// safety against '/' etc
+		primaryKey[i] = url.QueryEscape(keyVal)
+	}
+	indexPath := filepath.Join(string(assert.Type()), filepath.Join(primaryKey...))
+	_, err = db.statEntry(assertionsRoot, indexPath)
+	switch {
+	case err == nil:
+		// directory for assertion is present
+		curRevStr, err := db.readlinkEntry(assertionsRoot, indexPath, "latest")
+		if err != nil {
+			return fmt.Errorf("broken assertion store, reading 'latest' revision symlink: %v", err)
+		}
+		curRev, err := strconv.Atoi(curRevStr)
+		if err != nil {
+			return fmt.Errorf("broken assertion store, extracting revision from 'latest' revision symlink: %v", err)
+		}
+		if curRev >= assert.Revision() {
+			return ErrAssertionNotSuperseding
```

Comment:
I generally try to wait for a use case before making things a variable. Is there a clear reason why we'll want to test for this error, specifically, or is it just like the other errors in that we can just forward the issue to the user when necessary?